File: frontend/src/api/system/permission.ts
import { http } from "@/utils/http";
import type { ApiResult } from "@/types/api";

export type PermissionItem = {
  id: number;
  parentId: number;
  name: string;
  permissionKey: string;
  type: string;
  sort: number;
  status: number;
  remark?: string;
  children?: PermissionItem[];
  createTime?: string;
};

/** 树形查询 */
export const getPermissionTree = () => {
  return http.request<ApiResult<PermissionItem[]>>(
    "get",
    "/api/system/permission/tree"
  );
};

/** 新增权限 */
export const createPermission = (data: Partial<PermissionItem>) => {
  return http.request<ApiResult<void>>("post", "/api/system/permission", {
    data
  });
};

/** 修改权限 */
export const updatePermission = (data: Partial<PermissionItem>) => {
  return http.request<ApiResult<void>>("put", "/api/system/permission", {
    data
  });
};

/** 删除权限 */
export const deletePermission = (id: number) => {
  return http.request<ApiResult<void>>(
    "delete",
    `/api/system/permission/${id}`
  );
};

File: backend/src/main/java/com/ewcp/entity/Permission.java
package com.ewcp.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("t_permission")
public class Permission {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long parentId;

    private String name;

    private String permissionKey;

    private String type;

    private Integer sort;

    private Integer status;

    private String remark;

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private List<Permission> children;
}
